src/core/events/types/common.py

```diff
@@ -60,28 +66,42 @@ def explain(self):
         return self.__doc__
 
 
+global event_id_count_lock
+event_id_count_lock = threading.Lock()
+event_id_count_lock.acquire()
 event_id_count = 0
+event_id_count_lock.release()
+
 """ Discovery/Hunting Events """
+
+
 class NewHostEvent(Event):
     def __init__(self, host, cloud=None):
+        event_id_count_lock.acquire()
         global event_id_count
```

Comment:
As elsewhere, I don't think you need to grab the lock just to declare the variable. Test to make sure I'm correct though!

Reply:
I think you are correct, I've added these on the "Using stricter locking "policy"" commit since I didn't understand why the code still has threading issues (when it "evolved" from 3/4 to 96/100 and I couldn't understand why it is not 100/100).

Reply:
-->
Changed & Tested, you were correct :)

Reply:
I don't think you need to grab the lock for setting event_id_count to 0 either, because it's only run once when the module is loaded